module_utils/network_lsr/argument_validator.py

```diff
@@ -1008,6 +1009,31 @@ def __init__(self):
         )
 
 
+class ArgValidator_DictEthtoolRing(ArgValidatorDict):
+    def __init__(self):
+        ArgValidatorDict.__init__(
+            self,
+            name="ring",
+            nested=[
+                ArgValidatorNum(
+                    "rx", val_min=0, val_max=UINT32_MAX, default_value=None
```

Comment:
at https://github.com/linux-system-roles/network/pull/355#discussion_r615212067, I suggest to fix ArgValidatorNum() to reject boolean values. In a separate commit too, and with a comment in the commit message that this bug fix potentially breaks previously "working" playbooks (but realistically, they were not working, because setting 0 or 1 was unlikely intended).

Reply:
Separated commit was created and test cases was also added.